tests/guest-tools/win/test_guest_tools_win.py

```diff
@@ -0,0 +1,346 @@
+    # vm shutdown is not usable yet (there's no tools)
+    vm.ssh(
+        [
+            "powershell.exe",
+            "-noprofile",
+            "-noninteractive",
+            "Stop-Computer",
+            "-Force",
+        ]
+    )
+    wait_for(vm.is_halted, "Shutdown VM")
+    snapshot = vm.snapshot()
+    yield (vm, snapshot)
+    snapshot.destroy(verify=True)
+
+
+@pytest.fixture
+def vm_instance(vm_prepared: tuple[VM, Snapshot]):
+    (vm, snapshot) = vm_prepared
+    vm.start()
+    wait_for_guest_start(vm)
+    yield vm
+    snapshot.revert()
+
+
+@pytest.mark.multi_vms
+@pytest.mark.usefixtures("windows_vm")
+class TestGuestToolsWindows:
+    @pytest.fixture(scope="class")
+    def vm_install(self, vm_prepared):
+        (vm, snapshot) = vm_prepared
+        vm.start()
+        wait_for_guest_start(vm)
+        install_cert_and_tools(vm, PowerAction.Reboot, check=False)
+        yield vm
+        snapshot.revert()
+
+    def test_tools_after_reboot(self, vm_install: VM):
+        vm = vm_install
+        assert is_drivers_installed(vm)
+
+    def test_drivers_detected(self, vm_install: VM):
+        vm = vm_install
+        assert vm.param_get("PV-drivers-detected")
+
+
+@pytest.mark.multi_vms
+@pytest.mark.usefixtures("windows_vm")
+class TestGuestToolsWindowsDestructive:
+    @pytest.fixture
+    def vm_install(self, vm_instance: VM):
+        install_cert_and_tools(vm_instance, PowerAction.Nothing)
+        return vm_instance
+
+    @pytest.fixture
+    def vm_install_citrix_tools(self, vm_instance: VM):
+        install_other_drivers(
+            vm_instance,
+            "citrix-9.4.0\\managementagent-9.4.0-x64.msi",
+            is_msi=True,
+        )
+        return vm_instance
+
+    @pytest.fixture
+    def vm_install_xcpng_v8_tools(self, vm_instance: VM):
+        install_other_drivers(
+            vm_instance,
+            "xcp-ng-8.2.2.200\\managementagentx64.msi",
+            is_msi=True,
+        )
+        return vm_instance
+
+    @pytest.fixture
+    def vm_install_xcpng_v9_tools(self, vm_instance: VM):
+        vm = vm_instance
+
+        vm.insert_cd("other-guest-tools-win.iso")
+        time.sleep(5)
+
+        logging.info("Install VM root certs")
+        vm.ssh(
+            [
+                "certutil",
+                "-addstore",
+                "-f",
+                "Root",
+                "D:/xcp-ng-9.0.9000/testsign/XCP-ng_Test_Signer.crt",
+            ]
+        )
+        vm.ssh(
+            [
+                "certutil",
+                "-addstore",
+                "-f",
+                "TrustedPublisher",
+                "D:/xcp-ng-9.0.9000/testsign/XCP-ng_Test_Signer.crt",
+            ]
+        )
+
+        vm.eject_cd()
+
+        install_other_drivers(
+            vm,
+            "xcp-ng-9.0.9000\\package\\XenDrivers-x64.msi",
+            is_msi=True,
+        )
+        return vm
+
+    @pytest.fixture
+    def vm_install_wu_drivers(self, vm_prepared: VM):
```

Comment:
It's not obvious that "wu" means "Windows Update". Expand the acronym or add a comment?